realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1022,11 +1151,15 @@ public boolean isValid() {
      * @throws java.lang.IllegalArgumentException if one or more arguments do not match class or field type.
      */
     public RealmQuery<E> lessThan(String fieldName, Date value) {
+        realm.checkIfValid();
+
         long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.DATE);
         this.query.lessThan(columnIndices, value);
         return this;
     }
 
+    // Less Than Or Equal To
```

Comment:
Yes. I think they are from a time where the IDEs/editors didn't have a function to show all the methods in a separate pane.